api/app/Models/Pool.php

```diff
@@ -162,6 +161,11 @@ protected static function boot()
         });
     }
 
+    public function newEloquentBuilder($query): PoolBuilder
```

Comment:
This is what actually binds the new builder to the model, right?  Could you add a comment describing this?